### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by koudis**
## Summary by CodeRabbit

* **New Features**
  * Introduced centralized configuration management for CMake projects using the user package registry.
  * Added automated example builds and cross-platform testing workflows for Linux, Windows, and macOS.
  * Provided example configurations and usage documentation for quick adoption.
  * Enabled installation and uninstallation of configuration packages with improved error handling and clear user feedback.

* **Documentation**
  * Added comprehensive README files for both the main project and example usage, including setup instructions and function descriptions.

* **Tests**
  * Implemented extensive test coverage for configuration management, including error handling, variable retrieval, system isolation, and edge cases across platforms.

### CODE REVIEW

File: test/test_cases/fail/unix_home_missing/test_config/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    CMAKE_MINIMUM_REQUIRED(VERSION 3.22)
+    PROJECT(CMCONF_UNIX_HOME_MISSING_TEST_CONFIG)
+ENDIF()
+
+FIND_PACKAGE(CMLIB REQUIRED)
+
+INCLUDE("${CMAKE_CURRENT_LIST_DIR}/../../../../../CMCONF.cmake")
+
+SET(ENV{HOME} "")
+SET(CMCONF_UNINSTALL ON CACHE BOOL "" FORCE)
```
[the remark this answers is not in the document]
Comment (koudis) [reply test/test_cases/fail/unix_home_missing/test_config/CMakeLists.txt:16]: Is this really ok? Will it work?  Verify with Cmake doc.

File: test/test_cases/fail/windows_userprofile_missing/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+## Main
+#
+# Test CMCONF_INIT_SYSTEM error when USERPROFILE is not set on Windows
+#
+
+IF(NOT DEFINED CMAKE_SCRIPT_MODE_FILE)
+    CMAKE_MINIMUM_REQUIRED(VERSION 3.22)
+    PROJECT(CMCONF_WINDOWS_USERPROFILE_MISSING_TEST)
+ENDIF()
+
```
[the remark this answers is not in the document]
Comment (koudis) [reply test/test_cases/fail/windows_userprofile_missing/CMakeLists.txt:10]: Thats fine it is adressed in main cmakelist

File: test/test_cases/fail/variable_not_defined/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+INCLUDE("${CMAKE_CURRENT_LIST_DIR}/../../../test_cmconf_helpers.cmake")
+INCLUDE("${CMAKE_CURRENT_LIST_DIR}/../../../config_templates/config_template.cmake")
+
+TEST_CMCONF_CHECK_AND_INSTALL_CONFIG("${CMCONF_TEST_CONFIG_FILE}" "TEST")
+
+TEST_RUN_AND_CHECK_OUTPUT("test_config"
+    FATAL_ERROR_MESSAGE "CMCONF\\\\[TEST\\\\] - Variable 'NONEXISTENT_VAR' is not defined in configuration.*for system 'TEST'"
+)
+
+TEST_CMCONF_UNINSTALL_CONFIG("${CMCONF_TEST_CONFIG_FILE}" "TEST")
```
[the remark this answers is not in the document]
Comment (koudis) [reply test/test_cases/fail/variable_not_defined/CMakeLists.txt:23]: Good idea but it will not work. I need to run in Project and Script mode, so...
